docs/usage.rst

```diff
@@ -0,0 +1,134 @@
+
+.. code-block:: bash
+
+   nix-shell <pika-root>/shell.nix
+
+The ``nixpgks`` version is not pinned and may break occasionally.
+
+Including in CMake projects
+---------------------------
+
+Once installed, pika can be used in a CMake project through the ``pika::pika`` target. Other ways of
+depending on pika are likely to work but are not officially supported.
+
+.. code-block:: cmake
+
+   find_package(pika REQUIRED)
+   add_executable(app main.cpp)
+   target_link_libraries(app PRIVATE pika::pika)
+
+.. _cmake_configuration:
+
+Customizing the pika installation
+=================================
+
+The most important CMake options are listed in :ref:`getting_started`. Below is a more complete list
+of CMake options you can use to customize the installation.
+
+- ``PIKA_WITH_MALLOC``: This defaults to ``mimalloc`` which requires mimalloc to be installed.  Can
+  be set to ``tcmalloc``, ``jemalloc``, ``mimalloc``, or ``system``. Setting it to ``system`` can be
+  useful in debug builds.
+- ``PIKA_WITH_CUDA``: Enable CUDA support.
+- ``PIKA_WITH_HIP``: Enable HIP support.
+- ``PIKA_WITH_MPI``: Enable MPI support.
+- ``PIKA_WITH_APEX``: Enable `APEX <todo>`_ support.
+- ``PIKA_WITH_TRACY``: Enable `Tracy <todo>`_ support.
+- ``PIKA_WITH_GENERIC_CONTEXT_COROUTINES``: Enable the use of Boost.Context for fiber context
+  switching.
+- ``PIKA_WITH_TESTS``: Enable tests. Tests can be built with ``cmake --build . --target tests`` and
+  run with ``ctest --output-on-failure``.
+- ``PIKA_WITH_EXAMPLES``: Enable examples. Binaries will be placed under ``bin`` in the build
+  directory.
+
+.. _pika_stdexec:
+
+Relation to std::execution and stdexec
+======================================
+
+When pika was first created as a fork of `HPX <https://github.com/STEllAR-GROUP/hpx>`_ in 2022
+stdexec was in its infancy. Because of this, pika contains an implementation of a subset of the
+earlier revisions of P2300. The main differences to stdexec and the proposed facilities are:
+
+- The pika implementation uses C++17 and thus does not make use of concepts or coroutines. This
+  allows compatibility with slightly older compiler versions and e.g. nvcc.
+- The pika implementation uses ``value_types``, ``error_types``, and ``sends_done`` instead of
+  ``completion_signatures`` in sender types, as in the `first 3 revisions of P2300
+  <https://wg21.link/p2300r3>`_.
+
+pika has an experimental CMake option ``PIKA_WITH_STDEXEC`` which can be enabled to use stdexec for
+the P2300 facilities. pika brings the ``stdexec`` namespace into ``pika::execution::experimental``,
+but provides otherwise no guarantees of interchangeable functionality. pika only implements a subset
+of the proposed sender algorithms which is why we recommend that you enable ``PIKA_WITH_STDEXEC``
+whenever possible. We plan to deprecate and remove the P2300 implementation in pika in favour of
+stdexec and/or standard library implementations.
```

Comment:
Is there any rough/indicative timeline users can expect?

Reply:
That's a good question. If there were one, would you want it written here?

I think we're realistically looking at 6-12 months if we get stdexec working cleanly in DLA-Future with all configurations, but not all configurations even compile yet so I'm wary to give an estimate here.

Reply:
My question was mostly out of personal curiosity. I think a `.. note::` with a very rough timeline might be nice, but if it's too early to tell I'd leave it out then.